Classes/core/entity/enemy/Boss-1.cpp

```diff
@@ -1,16 +1,38 @@
 #include "Boss-1.h"
 #include "../../map.h"
+#include <stdexcept>
 
 namespace towerdefence {
 namespace core {
 
+Boss1::Boss1(id::Id id, const timer::Clock &clk)
+        : Enemy(id), release_skill_(clk.with_period_sec(20)) {}
+
 void Boss1::on_tick(GridRef g) {
-    for (auto &grid : g.map.grids) {
-        grid.with_tower([this](std::unique_ptr<Tower> &tower) {
-            tower->add_buff({this->id, Buff::DEFAULT}, Buff::attack_speed(-30) & Buff::silent(true));
-        });
+    this->update_buff(g.clock());
+
+    if (g.clock().is_triggered(release_skill_)) {
```

Comment:
These codes are repeated. Maybe we can abstract them somehow.

Reply:
waiting for your abstract